src/core/analyzer.ts

```diff
@@ -0,0 +1,44 @@
+import { FollowLists, FollowMap } from './types';
+
+export enum FollowStatus {
+  /**
+   * Determines if the current user is **followed by** the lookup user.
+   * E.g.: Lookup user follows Bob.
+   */
+  IS_FOLLOWED_BY_LOOKUP_USER = 0b01,
+
+  /**
+   * Determine if the current user **is following** the lookup user.
+   * E.g.: Bob follows lookup user.
+   */
+  IS_FOLLOWING_LOOKUP_USER = 0b10,
+
+  /**
+   * Determines if the lookup user follows and is followed by the current user.
+   */
+  BOTH = IS_FOLLOWED_BY_LOOKUP_USER | IS_FOLLOWING_LOOKUP_USER
```

Comment:
Esse | é um operador OR? Não deveria ser um AND?

Reply:
Não porque nesse caso a operação _bitwise_ `OR` é desejada para criar um "estado" que junta as duas _flags_.

Por exemplo:

```
a = 0b01
b = 0b10

a | b // Resulta em: 0b11
```

[Um bom artigo para entender esse tipo de operação bitwise.](https://www.alanzucconi.com/2015/07/26/enum-flags-and-bitwise-operators/)